telemetry-aware-scheduling/deploy/cluster-api/docker/capi-docker.md

```diff
@@ -0,0 +1,160 @@
+
+  ```bash
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: monitoring
+  labels:
+    name: monitoring
+  ````
+
+Prepend the following to `prometheus-custom-metrics.yaml`:
+  ```bash
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: custom-metrics
+  labels:
+    name: custom-metrics
+  ```
+
+The custom metrics adapter and the TAS deployment require TLS to be configured with a certificate and key.
+Information on how to generate correctly signed certs in kubernetes can be found [here](https://github.com/kubernetes-sigs/apiserver-builder-alpha/blob/master/docs/concepts/auth.md).
+Files ``serving-ca.crt`` and ``serving-ca.key`` should be in the current working directory.
+
+Run the following:
+
+  ```bash
+  kubectl -n custom-metrics create secret tls cm-adapter-serving-certs --cert=serving-ca.crt --key=serving-ca.key -oyaml --dry-run=client > custom-metrics-tls-secret.yaml
+  kubectl -n default create secret tls extender-secret --cert=serving-ca.crt --key=serving-ca.key -oyaml --dry-run=client > tas-tls-secret.yaml
+  ```
+
+**Attention: Don't commit the TLS certificate and private key to any Git repo as it is considered bad security practice! Make sure to wipe them off your workstation after applying the relative Secrets to your cluster.**
+
+You also need the TAS manifests (Deployment, Policy CRD and RBAC accounts) and the extender's "configmapgetter"
+ClusterRole. We will join the TAS manifests together, so we can have a single ConfigMap for convenience:
+
+  ```bash
+  yq '.' ../tas-*.yaml > tas.yaml
+  ```
+
+4. Create and apply the ConfigMaps
+
+  ```bash
+  kubectl create configmap custom-metrics-tls-secret-configmap --from-file=./custom-metrics-tls-secret.yaml -o yaml --dry-run=client > custom-metrics-tls-secret-configmap.yaml
+  kubectl create configmap custom-metrics-configmap --from-file=./prometheus-custom-metrics.yaml -o yaml --dry-run=client > custom-metrics-configmap.yaml
+  kubectl create configmap prometheus-configmap --from-file=./prometheus.yaml -o yaml --dry-run=client > prometheus-configmap.yaml
+  kubectl create configmap prometheus-node-exporter-configmap --from-file=./prometheus-node-exporter.yaml -o yaml --dry-run=client > prometheus-node-exporter-configmap.yaml
+  kubectl create configmap tas-configmap --from-file=./tas.yaml -o yaml --dry-run=client > tas-configmap.yaml
+  kubectl create configmap tas-tls-secret-configmap --from-file=./tas-tls-secret.yaml -o yaml --dry-run=client > tas-tls-secret-configmap.yaml
+  kubectl create configmap extender-configmap --from-file=../extender-configuration/configmap-getter.yaml -o yaml --dry-run=client > extender-configmap.yaml
+  ```
+
+Apply to the management cluster:
+
+  ```bash
+  kubectl apply -f '*-configmap.yaml'
+  ```
+
+5. Apply the ClusterResourceSets
+
+ClusterResourceSets resources are already given to you in `clusterresourcesets.yaml`.
+Apply them to the management cluster with `kubectl apply -f clusterresourcesets.yaml`
```

Comment:
Does this refer to docker/clusterresourcesets.yaml? And for generic, it also refers to it's own clusterresourcesets.yaml

Reply:
The CRS resources are actually the same. Do you think it would make more sense to have only one file for both vs duplicating it, if yes any idea where to put the CRS file in the folder tree?

Reply:
I think having the shared folder and just referencing the files is a good idea. Do we still need the clustersharedresources.yaml file in both (docker, generic) folders?